src/views/PayloadsView.vue

```diff
@@ -0,0 +1,115 @@
+<script setup>
+import { inject, onMounted, computed } from "vue";
+import { storeToRefs } from "pinia";
+
+import { useCoreDisplayStore } from "@/stores/coreDisplayStore";
+import UploadModal from "@/components/payloads/UploadModal.vue";
+import { useAbilityStore } from "@/stores/abilityStore";
+
+const $api = inject("$api");
+
+const abilityStore = useAbilityStore();
+const coreDisplayStore = useCoreDisplayStore();
+const { payloads } = storeToRefs(abilityStore);
+const { modals } = storeToRefs(coreDisplayStore);
+
+const structuredPayloads = computed(() => {
+  const regex = /^(?:plugins\/([^/]+)|data)?\/payloads\/(.+)$/;
+  return payloads.value.map(payload => {
+    const match = payload.match(regex);
+    if (!match) {
+      console.error(`Payload path "${payload}" does not match the expected format.`);
+      return null;
+    }
+    let belongsToAPlugin = match[1] !== undefined;
```

Comment:
Just curious, why is this line needed? Isnt it just defined below in the returning dict?

Reply:
Good question.
As I said previously, I decided not to break the initial payload GET endpoint because it is being used by the ability view (when adding/modifying an ability).
This endpoint returns a list of strings, not a dictionary.
Note the swagger is wrong (**I can update it easily to the "true" returned data structure if you ask for it**).

Example of returned payloads:
```json
["payload1.exe", "payload2.sh"]
```

In the API, i just added _optional_ parameters that change the returned string (adding path, excluding plugins, sorting), but those parameters still do not return a proper list of dictionaries (which would be actually the ideal way to structure data).

So, at the frontend side, I managed to deduct (rebuild) such a structure, hence the code.

This is not ideal, but the GET payload endpoint breaking change could be handled in a different topic/issue/PR.

Just tell me what you prefer:

* validate this PR (no breaking change of the API)
* or request a change: Breaking GET payload API returning a list of structured data, implies less frontend boilerplate but the ability add/update view is to be updated.

Reply:
Im okay with the current implementation if you are, and approve of merge. 

Im also okay with changing this in future set of PRs. *